#import libraries

In [1]:
import pandas as pd
import numpy as np
import os

#import dataframe

In [2]:
path = r'C:\Instacart Basket Achievement 4 NEW'

In [3]:
path

'C:\\Instacart Basket Achievement 4 NEW'

In [4]:
df_orders_products_combined = pd.read_pickle(os.path.join(path, '02 Data' , 'Prepared Data', 'orders_products_combined.pkl'))

In [5]:
df_orders_products_combined.shape

(32434489, 11)

In [6]:
df_orders_products_combined

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0                  0   2539329        1             1                   2   
1                  0   2539329        1             1                   2   
2                  0   2539329        1             1                   2   
3                  0   2539329        1             1                   2   
4                  0   2539329        1             1                   2   
...              ...       ...      ...           ...                 ...   
32434484     3421081   2977660   206209            13                   1   
32434485     3421081   2977660   206209            13                   1   
32434486     3421081   2977660   206209            13                   1   
32434487     3421081   2977660   206209            13                   1   
32434488     3421081   2977660   206209            13                   1   

          order_hour_of_day  days_since_prior_order  product_id  \
0                         8                     NaN         196   
1                         8                     NaN       14084   
2                         8                     NaN       12427   
3                         8                     NaN       26088   
4                         8                     NaN       26405   
...                     ...                     ...         ...   
32434484                 12                     7.0       14197   
32434485                 12                     7.0       38730   
32434486                 12                     7.0       31477   
32434487                 12                     7.0        6567   
32434488                 12                     7.0       22920   

          add_to_cart_order  reordered _merge  
0                         1          0   both  
1                         2          0   both  
2                         3          0   both  
3                         4          0   both  
4                         5          0   both  
...                     ...        ...    ...  
32434484                  5          1   both  
32434485                  6          0   both  
32434486                  7          0   both  
32434487                  8          0   both  
32434488                  9          0   both  

[32434489 rows x 11 columns]

In [7]:
df_orders_products_combined['_merge'].value_counts() ## Checking value count of '_merge' in df_orders_products_combined before merging with df_prods

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

In [8]:
df_orders_products_combined = df_orders_products_combined.drop(columns = ['Unnamed: 0'])# Dropping the extra indexing column "Unnamed: 0"

In [9]:
df_orders_products_combined.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  
0                     NaN         196                  1          0   both  
1                     NaN       14084                  2          0   both  
2                     NaN       12427                  3          0   both  
3                     NaN       26088                  4          0   both  
4                     NaN       26405                  5          0   both

In [10]:
df_prods = pd.read_csv(os.path.join(path, '02 Data' , 'Prepared Data', 'products_checked_1.csv'), index_col=False)

In [11]:
df_prods.shape

(49672, 6)

In [12]:
df_prods.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [13]:
df_prods = df_prods.drop(columns = ['Unnamed: 0'])

In [14]:
df_prods.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [15]:
df_orders_products_merged = df_orders_products_combined.merge(df_prods, on = 'product_id') #merging dataframes

In [16]:
df_orders_products_merged.shape

(32404859, 14)

In [17]:
df_orders_products_merged.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  \
0                     NaN         196                  1          0   both   
1                    15.0         196                  1          1   both   
2                    21.0         196                  1          1   both   
3                    29.0         196                  1          1   both   
4                    28.0         196                  1          1   both   

  product_name  aisle_id  department_id  prices  
0         Soda        77              7     9.0  
1         Soda        77              7     9.0  
2         Soda        77              7     9.0  
3         Soda        77              7     9.0  
4         Soda        77              7     9.0

In [18]:
df_orders_products_merged['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

There were 32,434,489 rows in the orders_products_combined data frame. There were 49,672 rows in the prods data frame. When merged with an inner join the result was 32,404,859 rows. This is 29,630 fewer records than in the orders_products_combined data frame

In [20]:
df_orders_products_merged.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_merged.pkl'))